packages/executor/src/services/UserOpValidation/validators/safe.ts

```diff
@@ -80,13 +88,46 @@ export class SafeValidationService {
       .add(userOp.verificationGasLimit)
       .add(userOp.callGasLimit);
 
+    // if wallet is not created, trace simulateHandleOp to check that callData does not revert
+    if (ethers.constants.Zero.eq(userOp.nonce)) {
+      const simulateHandleOpTx: providers.TransactionRequest = {
+        to: entryPoint,
+        data: entryPointContract.interface.encodeFunctionData(
+          "simulateHandleOp",
+          [userOp, AddressZero, BytesZero]
+        ),
+        gasLimit: simulationGas.mul(3),
+        ...gasPrice,
+      };
+      const traceCall: BundlerCollectorReturn =
+        await this.gethTracer.debug_traceCall(simulateHandleOpTx);
+      for (const log of traceCall.logs) {
+        if (
+          log.topics[0] ===
+          "0x49628fd1471006c1482da88028e9ce4dbb080b815c9b0344d39e5a8e6ec1419f" // user operation event
```

Comment:
Is this specific Etherspot? What does this hash mean? Will this change for the new EP?

Reply:
it is the hash of UserOperationEvent - https://github.com/eth-infinitism/account-abstraction/blob/develop/contracts/interfaces/IEntryPoint.sol#L29C11-L37

It is from EntryPoint v6 and it will change for new EP

Reply:
Should we then make it configurable? Also we have to validate if the hash of the UserOpEvent is the same on different networks?

Reply:
>Also we have to validate if the hash of the UserOpEvent is the same on different networks?

Yes, it is same on all chains because it comes from EPv6's bytecode

>Should we then make it configurable

We can, but at the moment the bundler can only serve EPv6 anyway. We can make the hash configurable when we start working on supporting EPv7

Reply:
@ch4r10t33r yes, it's the same on both of those chains

On Linea: https://lineascan.build/tx/0x7708a4de198e15cc1f99dd70b453f9b02ad135ea2819530b389e003f7a3ba2d5#eventlog:~:text=0x49628fd1471006c1482da88028e9ce4dbb080b815c9b0344d39e5a8e6ec1419f

On Bifrost though the EntryPoint's address is different, the event hash is the same
https://explorer.mainnet.thebifrost.io/tx/0xe0ba8187785eb59e24c6c76bba7f23b75fe693f5169b3f89668c9e81fa6dbdb4/logs#:~:text=0x49628fd1471006c1482da88028e9ce4dbb080b815c9b0344d39e5a8e6ec1419f